src/optimizer/memo.cpp

```diff
@@ -14,34 +14,71 @@
 #include "optimizer/memo.h"
```

Comment:
It would be nice if you added more documentation to memo.h to give a better high-level idea of what these Memo objects are used for.